spec/requests/disks_spec.rb

```diff
@@ -0,0 +1,56 @@
+RSpec.describe "Disks API" do
+  describe "as a subcollection of VMs" do
+    describe "GET /api/vms/:c_id/disks" do
+      it "can list the snapshots of a VM" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :disks, :read, :get))
+        hw = FactoryBot.create(:hardware)
+        vm = FactoryBot.create(:vm_vmware, :hardware => hw)
+        disk = FactoryBot.create(:disk, :hardware => hw)
+
+        _other_disk = FactoryBot.create(:disk)
+
+        get(api_vm_disks_url(nil, vm))
+
+        expected = {
+          "count"     => 2,
+          "name"      => "disks",
+          "subcount"  => 1,
+          "resources" => [
+            {"href" => api_vm_disk_url(nil, vm, disk)}
+          ]
+        }
+
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    describe "GET /api/vms/:c_id/disks/:s_id" do
+      it "can show a VM's disk" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :disks, :read, :get))
+        hw = FactoryBot.create(:hardware)
+        vm = FactoryBot.create(:vm_vmware, :hardware => hw)
+        disk = FactoryBot.create(:disk, :hardware => hw)
+
+        get(api_vm_disk_url(nil, vm, disk))
+
+        expected = {
+          "href"              => api_vm_disk_url(nil, vm, disk),
+          "id"                => disk.id.to_s,
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "will not show a disk unless authorized" do
```

Comment:
@lpichler @abellotti I'm not sure how RBAC works here, maybe I'm missing something but this test is failing on it. Could you please help me out?

Reply:
@skateman you need to define also `disks_subresource_actions` because as I understand you are not fetching collection of disk - then also specs will pass.

thanks


```
diff --git a/config/api.yml b/config/api.yml
index 0269ba6b..f650396a 100644
--- a/config/api.yml
+++ b/config/api.yml
@@ -3908,6 +3908,11 @@
       - :name: read
         :identifier:
         - vm_show
+    :disks_subresource_actions:
+      :get:
+        - :name: read
+          :identifier:
+            - vm_show
     :tags_subcollection_actions:
       :post:
       - :name: assign
```

Reply:
Thanks, except the indentation in your code is wrong, but it seems working after I fixed it.